server/sonar-web/src/main/js/apps/overview/main/enhance.tsx

```diff
@@ -41,24 +42,44 @@ import {
   getComponentIssuesUrl,
   getMeasureHistoryUrl
 } from '../../../helpers/urls';
+import { Component } from '../../../app/types';
+import { History } from '../../../api/time-machine';
+
+export interface EnhanceProps {
+  branch?: string;
+  component: Component;
+  measures: MeasureEnhanced[];
+  leakPeriod?: { index: number; date?: string };
+  history?: History;
+  historyStartDate?: Date;
+}
+
+export interface ComposedProps extends EnhanceProps {
+  getValue: (measure: MeasureEnhanced) => string | undefined;
+  renderHeader: (domain: string, label: string) => React.ReactNode;
+  renderMeasure: (metricKey: string) => React.ReactNode;
+  renderMeasureVariation: (metricKey: string, customLabel: string) => React.ReactNode;
+  renderRating: (metricKey: string) => React.ReactNode;
+  renderIssues: (metric: string, type: string) => React.ReactNode;
+  renderHistoryLink: (metricKey: string) => React.ReactNode;
+  renderTimeline: (metricKey: string, range: string, children?: React.ReactNode) => React.ReactNode;
+}
 
-export default function enhance(ComposedComponent) {
-  return class extends React.PureComponent {
+export default function enhance(ComposedComponent: React.ComponentType<ComposedProps>) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this function to reduce its Cognitive Complexity from 28 to the 15 allowed. ([typescript:S3776](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3776))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fgreg%2Foverview_to_ts&issues=AWEED0Avf6JiG_HArpbx&open=AWEED0Avf6JiG_HArpbx)

[ISSUE_KEY]: AWEED0Avf6JiG_HArpbx

Reply:
This issue has been resolved by a user in SonarQube

[ISSUE_STATUS]: RESOLVED